src/java.desktop/share/classes/sun/java2d/cmm/lcms/LCMS.java

```diff
@@ -80,8 +86,13 @@ private static LCMSProfile getLcmsProfile(Profile p) {
     }
 
     @Override
-    public synchronized void setTagData(Profile p, int tagSignature, byte[] data) {
-        getLcmsProfile(p).setTag(tagSignature, data);
+    public void setTagData(Profile p, int tagSignature, byte[] data) {
+        long stamp = lock.writeLock();
+        try {
+            getLcmsProfile(p).setTag(tagSignature, data);
+        } finally {
+            lock.unlockWrite(stamp);
+        }
```

Comment:
We have only one LCMS object at a time, so using "synchronized" and "writeLock" are kind of equivalent.